data-prepper-plugins/cloudwatch-logs/src/main/java/org/opensearch/dataprepper/plugins/sink/threshold/ThresholdCheck.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.sink.threshold;
+/**
+ * ThresholdCheck receives paramaters for which to reference the
+ * limits of a buffer and CloudWatchLogsClient before making a
+ * PutLogEvent request to AWS.
+ */
+public class ThresholdCheck {
+    private final int batchSize;
+    private final int maxEventSizeBytes;
+    private final int maxRequestSizeBytes;
+    private final long logSendInterval;
+
+    public ThresholdCheck(final int batchSize, final int maxEventSizeBytes, final int maxRequestSizeBytes, final int logSendInterval) {
+        this.batchSize = batchSize;
+        this.maxEventSizeBytes = maxEventSizeBytes;
+        this.maxRequestSizeBytes = maxRequestSizeBytes;
+        this.logSendInterval = logSendInterval;
+    }
+
+    /**
+     * Checks to see if we exceed any of the threshold conditions.
+     * @param currentTime - (long) denoting the time in seconds.
+     * @param currentRequestSize - size of request in bytes.
+     * @param batchSize - size of batch in events.
+     * @return boolean - true if we exceed the threshold events or false otherwise.
+     */
+    public boolean isGreaterThanThresholdReached(final long currentTime, final int currentRequestSize, final int batchSize) {
+        return ((checkGreaterThanBatchSize(batchSize) || checkLogSendInterval(currentTime)
+                || checkGreaterThanMaxRequestSize(currentRequestSize)) && (batchSize > 0));
+    }
+
+    /**
+     * Checks to see if we equal any of the threshold conditions.
+     * @param currentRequestSize - size of request in bytes.
+     * @param batchSize - size of batch in events.
+     * @return boolean - true if we equal the threshold events or false otherwise.
+     */
+    public boolean isEqualToThresholdReached(final int currentRequestSize, final int batchSize) {
+        return ((checkEqualBatchSize(batchSize) || checkEqualMaxRequestSize(currentRequestSize)) && (batchSize > 0));
+    }
+
+    /**
+     * Checks if the interval passed in is equal to or greater
+     * than the threshold interval for sending PutLogEvents.
+     * @param currentTimeSeconds int denoting seconds.
+     * @return boolean - true if greater than or equal to logInterval, false otherwise.
+     */
+    private boolean checkLogSendInterval(final long currentTimeSeconds) {
```

Comment:
nit: would renaming the `check_` methods to match their description and return types. It sounds like a validation method but its just a comparison.

For example `checkGreaterThanMaxEventSize` -> `isGreaterThanMaxEventSize`

Reply:
Got it, I have made the changes for the next PR update.